pkg/cnab/provider/parameters.go

```diff
@@ -51,20 +50,21 @@ func (d *Duffle) loadParameters(claim *claim.Claim, rawOverrides map[string]stri
 	// *but* should not apply to this action.
 	// When this occurs, we set an override to the current value in the claim such that
 	// bundle.ValuesOrDefaults does not return an error
-	// TODO: will have to refactor loop once required is back as a bool on the ParameterDefinition
-	for _, required := range bun.Parameters.Required {
-		if _, exists := rawOverrides[required]; !exists {
-			if !appliesToAction(action, bun.Parameters.Fields[required]) {
-				overrides[required] = claim.Parameters[required]
+	for name, param := range bun.Parameters {
```

Comment:
This logic makes my head hurt...  Am I reading it right that we want to populate overrides with the value from the claim when the parameter is required but the user didn't supply a value?

I'm worried that we may not have adequate test coverage for this code here. But that was the case before this code was changed so cest le vie! 😂

Reply:
> Am I reading it right that we want to populate overrides with the value from the claim when the parameter is required but the user didn't supply a value?

That is indeed the case.  Agreed, it's not very understandable logic.  This came in with https://github.com/deislabs/porter/pull/476 and included a test in https://github.com/deislabs/porter/blob/2a1d699b49b628fbc68a2bfdd52428b9a5434ed3/pkg/cnab/provider/parameters_test.go#L157-L191

However, perhaps we can create a work item to revisit?

Reply:
Let's just bury the body and back away slowly. Thanks for digging up the test to make me feel better! 🙌

Reply:
:joy: